src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas2/Cas2ApplicationTest.kt

```diff
@@ -179,6 +183,112 @@ class Cas2ApplicationTest : IntegrationTestBase() {
+        Pair("Awaiting arrival", UUID.fromString("89458555-3219-44a2-9584-c4f715d6b565")),
+      )
+
+      val applicationSchema = cas2ApplicationJsonSchemaEntityFactory.produceAndPersist {
+        withAddedAt(OffsetDateTime.now())
+        withId(UUID.randomUUID())
+      }
+
+      fun createApplication(userEntity: NomisUserEntity, offenderDetails: OffenderDetailSummary): Cas2ApplicationEntity {
+        return cas2ApplicationEntityFactory.produceAndPersist {
+          withApplicationSchema(applicationSchema)
+          withCreatedByUser(userEntity)
+          withCrn(offenderDetails.otherIds.crn)
+          withCreatedAt(OffsetDateTime.now().minusDays(28))
+          withConditionalReleaseDate(LocalDate.now().plusDays(1))
+        }
+      }
+
+      fun createStatusUpdate(status: Pair<String, UUID>, application: Cas2ApplicationEntity) {
+        val status = cas2StatusUpdateEntityFactory.produceAndPersist {
+          withLabel(status.first)
+          withStatusId(status.second)
+          withApplication(application)
+          withAssessor(externalUserEntityFactory.produceAndPersist())
+        }
+      }
+
+      fun unexpiredDateTime() = OffsetDateTime.now().randomDateTimeBefore()
+      fun expiredDateTime() = unexpiredDateTime().minusDays(14)
+
+      val unexpiredDateTimesAllStatuses = List(unexpiredSubset.union(expiredSubset).size) { unexpiredDateTime() }
+      val unexpiredDateTimesUnexpiredSubset = List(unexpiredSubset.size) { unexpiredDateTime() }
+      val expiredDateTimesExpiredSubset = List(expiredSubset.size) { expiredDateTime() }
+
+      val unexpiredApplicationIds = mutableSetOf<UUID>()
+      val expiredApplicationIds = mutableSetOf<UUID>()
+
+      `Given a CAS2 POM User` { userEntity, jwt ->
+        `Given an Offender` { offenderDetails, _ ->
+
+          repeat(2) {
+            unexpiredApplicationIds.add(createApplication(userEntity, offenderDetails).id)
+          }
+
+          mockkStatic(OffsetDateTime::class)
+
+          unexpiredSubset.union(expiredSubset).forEachIndexed { index, status ->
+            val application = createApplication(userEntity, offenderDetails)
+            every { OffsetDateTime.now() } returns unexpiredDateTimesAllStatuses[index]
+            createStatusUpdate(status, application)
+            unexpiredApplicationIds.add(application.id)
+          }
+
+          unexpiredSubset.forEachIndexed { index, status ->
+            val application = createApplication(userEntity, offenderDetails)
+            every { OffsetDateTime.now() } returns unexpiredDateTimesUnexpiredSubset[index]
+            createStatusUpdate(status, application)
+            unexpiredApplicationIds.add(application.id)
+          }
+
+          expiredSubset.forEachIndexed { index, status ->
+            val application = createApplication(userEntity, offenderDetails)
+            every { OffsetDateTime.now() } returns expiredDateTimesExpiredSubset[index]
```

Comment:
sorry, I'm finding all the date time stuff and the unions quite hard to parse...
are you mocking `OffsetDateTime.now()` here because it is called in the status update factory? and you want to set the status update's `createdAt` to a date that is 14 days or more in the past?

Would it be possible to ditch all these lists of date times and this mocking and just say what date you want status update to be have as the `createdAt` ?

```


      fun createStatusUpdate(status: Pair<String, UUID>, application: Cas2ApplicationEntity, createdAt: OffsetDateTime) {
        val status = cas2StatusUpdateEntityFactory.produceAndPersist {
          withLabel(status.first)
          withStatusId(status.second)
          withApplication(application)
          withAssessor(externalUserEntityFactory.produceAndPersist())
          withCreatedAt(createdAt)
        }
      }
 ```
 
 I think we generally only mock a return when its being used within the code under test, not to setup the testing scenario

Reply:
this can be ignored - should have read the commit message 🙃